### ISSUE TITLE
[release-1.22] pull-kubernetes-integration is failing

### ISSUE THREAD

**Issue by ialidzhikov** [kubernetes/kubernetes#105436]
#### Which jobs are failing:

pull-kubernetes-integration

See the runs related to https://github.com/kubernetes/kubernetes/pull/105139 - https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=105139

#### Which test(s) are failing:

Not clear. Even for some of the runs I don't see a particular test failure.

#### Since when has it been failing:

Not clear.

#### Testgrid link:

See the runs related to https://github.com/kubernetes/kubernetes/pull/105139 - https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=105139

#### Reason for failure:

Not clear.

#### Relevant SIG

/sig testing

**Comment by ehashman** [comment]
/priority critical-urgent
/triage accepted

https://storage.googleapis.com/k8s-triage/index.html?ci=0&pr=1&job=pull-kubernetes-integration

https://github.com/kubernetes/kubernetes/pull/105452#issuecomment-933657752

**Comment by spiffxp** [comment]
Tough to confirm when this started since there's no easy mechanism to filter history for presubmits by base branch

I'm trying to script something together to scrape GCS, so far I see the following have had failed pull-kubernetes-integration runs
- https://github.com/kubernetes/kubernetes/pull/105139
- https://github.com/kubernetes/kubernetes/pull/105366
- https://github.com/kubernetes/kubernetes/pull/105444
- https://github.com/kubernetes/kubernetes/pull/105452

**Comment by spiffxp** [comment]
Maybe related, we saw the presubmit timing out a bunch back in July, seems like that mostly stopped by 2020-07-09.  But we never closed out investigating, and I suspect these are timeouts based on spyglass lying (it's UI shows pass but the job is failing), ref: https://github.com/kubernetes/kubernetes/issues/103512

**Comment by ehashman** [comment]
Cold this be networking related?

```
W1004 19:01:13.392630  118232 reflector.go:441] k8s.io/client-go/informers/factory.go:134: watch of *v1beta1.PriorityLevelConfiguration ended with: an error on the server ("unable to decode a
n event from the watch stream: http2: client connection lost") has prevented the request from succeeding
E1004 19:01:07.379813  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.Endpoints: Get "https://127.0.0.1:44191/api/v1/endpoints?allowWatchBookmarks=tru
e&resourceVersion=19603&timeout=6m30s&timeoutSeconds=390&watch=true": http2: client connection lost
E1004 19:01:22.793215  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.ServiceAccount: Get "https://127.0.0.1:42523/api/v1/serviceaccounts?allowWatchBo
okmarks=true&resourceVersion=20170&timeout=5m19s&timeoutSeconds=319&watch=true": http2: client connection lost
E1004 19:01:16.434912  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.ServiceAccount: Get "https://127.0.0.1:44191/api/v1/serviceaccounts?allowWatchBo
okmarks=true&resourceVersion=19610&timeout=7m57s&timeoutSeconds=477&watch=true": http2: client connection lost
W1004 19:01:14.622372  118232 reflector.go:441] k8s.io/kube-aggregator/pkg/client/informers/externalversions/factory.go:117: watch of *v1.APIService ended with: an error on the server ("unabl
e to decode an event from the watch stream: http2: client connection lost") has prevented the request from succeeding
E1004 19:01:37.310019  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.Pod: Get "https://127.0.0.1:35693/api/v1/pods?allowWatchBookmarks=true&resourceV
ersion=24561&timeout=5m33s&timeoutSeconds=333&watch=true": http2: client connection lost
E1004 19:01:25.106278  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.ValidatingWebhookConfiguration: Get "https://127.0.0.1:43013/apis/admissionregis
tration.k8s.io/v1/validatingwebhookconfigurations?allowWatchBookmarks=true&resourceVersion=18618&timeout=5m3s&timeoutSeconds=303&watch=true": http2: client connection lost
E1004 19:01:33.548156  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.Service: Get "https://127.0.0.1:37093/api/v1/services?allowWatchBookmarks=true&r
esourceVersion=21826&timeout=6m39s&timeoutSeconds=399&watch=true": http2: client connection lost
E1004 19:01:33.127381  118232 reflector.go:138] k8s.io/kubernetes/pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go:444: Failed to watch *v1.Co
nfigMap: Get "https://127.0.0.1:45715/api/v1/namespaces/kube-system/configmaps?allowWatchBookmarks=true&resourceVersion=818&timeout=8m30s&timeoutSeconds=510&watch=true": http2: client connect
ion lost
E1004 19:01:43.283532  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.ValidatingWebhookConfiguration: Get "https://127.0.0.1:37093/apis/admissionregis
tration.k8s.io/v1/validatingwebhookconfigurations?allowWatchBookmarks=true&resourceVersion=21114&timeout=7m16s&timeoutSeconds=436&watch=true": http2: client connection lost
E1004 19:01:41.620012  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.MutatingWebhookConfiguration: Get "https://127.0.0.1:43031/apis/admissionregistr
ation.k8s.io/v1/mutatingwebhookconfigurations?allowWatchBookmarks=true&resourceVersion=19493&timeout=7m1s&timeoutSeconds=421&watch=true": http2: client connection lost
E1004 19:01:49.585794  118232 reflector.go:138] k8s.io/kube-aggregator/pkg/client/informers/externalversions/factory.go:117: Failed to watch *v1.APIService: Get "https://127.0.0.1:43013/apis/
apiregistration.k8s.io/v1/apiservices?allowWatchBookmarks=true&resourceVersion=18814&timeout=7m42s&timeoutSeconds=462&watch=true": http2: client connection lost
W1004 19:01:42.463442  118232 reflector.go:441] k8s.io/client-go/informers/factory.go:134: watch of *v1.Namespace ended with: an error on the server ("unable to decode an event from the watch
 stream: http2: client connection lost") has prevented the request from succeeding
E1004 19:01:45.587331  118232 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.ServiceAccount: Get "https://127.0.0.1:39851/api/v1/serviceaccounts?allowWatchBo
okmarks=true&resourceVersion=13750&timeout=5m17s&timeoutSeconds=317&watch=true": http2: client connection lost
W1004 19:01:59.024099  118232 reflector.go:441] k8s.io/client-go/informers/factory.go:134: watch of *v1beta1.FlowSchema ended with: an error on the server ("unable to decode an event from the
 watch stream: http2: client connection lost") has prevented the request from succeeding
W1004 19:02:01.132622  118232 clientconn.go:1326] [core] grpc: addrConn.createTransport failed to connect to {127.0.0.1:2379 127.0.0.1 <nil> 0 <nil>}. Err: connection error: desc = "transport
: Error while dialing dial tcp 127.0.0.1:2379: i/o timeout". Reconnecting...
W1004 19:01:54.388525  118232 reflector.go:441] k8s.io/client-go/informers/factory.go:134: watch of *v1.StorageClass ended with: an error on the server ("unable to decode an event from the wa
tch stream: http2: client connection lost") has prevented the request from succeeding
SIGQUIT: quit
PC=0x475ee1 m=0 sigcode=0

<giant stacktrace>
```

from https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/105452/pull-kubernetes-integration/1445094410348924928/build-log.txt (note filesize is 114MB)

**Comment by ehashman** [comment]
It looks like `=== FAIL: test/integration/apiserver/admissionwebhook TestWebhookTimeoutWithoutWatchCache (unknown)` is timing out and everything afterwards is breaking

[build-log.txt](https://github.com/kubernetes/kubernetes/files/7281093/build-log.txt)

**Comment by ehashman** [comment]
/sig api-machinery
cc @kubernetes/sig-api-machinery-bugs

**Comment by spiffxp** [comment]
tl;dr suggest trying to see what changed ~2021-09-27 since that seems to be around when the job stopped passing

Scraped last 800 runs of pull-kubernetes-integration, filtered down to those that ran against release-1.22.  First few entries from that list
```
1440397390203850752, 1632252591, 1632254857, SUCCESS, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105154, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105154/pull-kubernetes-integration/1440397390203850752
1440505394534289408, 1632278338, 1632280852, FAILURE, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105137, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105137/pull-kubernetes-integration/1440505394534289408
1440676558653100032, 1632319159, 1632321510, SUCCESS, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105137, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105137/pull-kubernetes-integration/1440676558653100032
1440911029000212480, 1632375145, 1632376170, FAILURE, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105139, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105139/pull-kubernetes-integration/1440911029000212480
1442364975170457600, 1632721783, 1632724274, FAILURE, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/104988, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/104988/pull-kubernetes-integration/1442364975170457600
1442415842519158784, 1632733833, 1632736174, SUCCESS, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/104988, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/104988/pull-kubernetes-integration/1442415842519158784
1442486689749536768, 1632750738, 1632752926, SUCCESS, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105277, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105277/pull-kubernetes-integration/1442486689749536768
1442958977880035328, 1632863313, 1632865629, FAILURE, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105139, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105139/pull-kubernetes-integration/1442958977880035328
1442974113156042752, 1632867067, 1632869682, FAILURE, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105331, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105331/pull-kubernetes-integration/1442974113156042752
1443111120528216064, 1632899587, 1632902313, FAILURE, presubmit, release-1.22, https://github.com/kubernetes/kubernetes/pull/105139, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105139/pull-kubernetes-integration/1443111120528216064
```

https://github.com/kubernetes/kubernetes/pull/105277 had the last successful run, https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105277/pull-kubernetes-integration/1442486689749536768, started at `2021-09-27 13:52:18 +0000 UTC`

https://github.com/kubernetes/kubernetes/pull/104988 is the closest PR to merge to that at `Mon Sep 27 02:51:36 2021 -0700`, but if I understand UTC offsets correctly, that is a few hours _before_ the last successful run

https://github.com/kubernetes/kubernetes/pull/105154 is the next PR but appears unrelated, updates storage e2es

https://github.com/kubernetes/kubernetes/pull/105139 has the first failing run,  https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105139/pull-kubernetes-integration/1442958977880035328 started at `2021-09-28 21:08:33 +0000 UTC`

**Comment by liggitt** [comment]
https://testgrid.k8s.io/sig-release-1.22-blocking#integration-1.22&width=5&graph-metrics=test-duration-minutes shows test variance increasing a lot around 9/27 - 9/28 timeframe

**Comment by liggitt** [comment]
comparing the last successful and first failed job's podinfo, the first failed job has:

```
				"prow.k8s.io/retest": "true",
```

what does that do?

**Comment by spiffxp** [comment]
@liggitt I think that's the integration tests becoming less I/O limited, that was around the time we migrated k8s-infra-prow-build to a local-ssd-backed nodepool, ref: https://github.com/kubernetes/k8s.io/issues/1187#issuecomment-929466293

**Comment by liggitt** [comment]
> ```
> 				"prow.k8s.io/retest": "true",
> ```
> 
> 
>     
>   
> 
> what does that do?

nm, just records whether it was triggered by a retest (https://github.com/kubernetes/test-infra/blob/master/prow/kube/prowjob.go#L64-L65)

**Comment by liggitt** [comment]
kubekins jumped from gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210917-ee1e7c845b-1.22 to gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210928-2a55334641-1.22

no other notable changes between the job podinfos

**Comment by spiffxp** [comment]
- https://github.com/kubernetes/test-infra/pull/23775 was the bump away from `gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210917-ee1e7c845b-1.22` with changes: https://github.com/kubernetes/test-infra/compare/ee1e7c845b...471c27b6e3
- https://github.com/kubernetes/test-infra/pull/23795 was the bump to `gcr.io/k8s-staging-test-infra/kubekins-e2e:v20210928-2a55334641-1.22` with changes: https://github.com/kubernetes/test-infra/compare/6525112548...2a55334641
- that makes full range of changes: https://github.com/kubernetes/test-infra/compare/ee1e7c845b...2a55334641 (oh good, those SHA's match up, that would be a faster way of doing it next time...)

**Comment by liggitt** [comment]
opened PR with the last-merged PR to release-1.22 reverted at https://github.com/kubernetes/kubernetes/pull/105473

I'm not sure how that would have passed the presubmit to get in if that was the issue, but just to double check

**Comment by spiffxp** [comment]
Looking through https://github.com/kubernetes/test-infra/compare/ee1e7c845b...2a55334641 I'm pretty sure https://github.com/kubernetes/test-infra/pull/23757 would have found its way into the image, which mangles iptables a bit if docker-in-docker is enabled.

I'm not sure that actually takes effect for this job though, and I'd be confused why it only affects the release-1.22 branch.

**Comment by spiffxp** [comment]
It does take effect, I see `configure iptables to set MTU` at the top of https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105139/pull-kubernetes-integration/1442958977880035328

The Google Cloud incident that motivated this change is over (https://github.com/kubernetes/test-infra/issues/23741), so theoretically we should be safe to revert this change.  I had been dragging my feet because I didn't have confidence we weren't going to run into it again:

**Comment by spiffxp** [comment]
Another angle to take: why is it passing in CI but failing in presubmit?
- The CI job has a memory limit of 20Gi where the presubmit has a memory limit of 15Gi
- The CI job runs with env `SHORT=--short=false`

**Comment by spiffxp** [comment]
Well, it's not related to the MTU workaround I proposed in https://github.com/kubernetes/kubernetes/issues/105436#issuecomment-933874821.  https://github.com/kubernetes/kubernetes/pull/105488 shows integration failing with and without that in place:
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105488/pull-kubernetes-integration-1-22-canary/1445421649049423872
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105488/pull-kubernetes-integration/1445421415124701184

**Comment by spiffxp** [comment]
At this point I have run out of "what changed" stuff to hunt down for the job as a whole.  Need help figuring out what tests are actually failing and why.

Tests that failed in regular integration
```
k8s.io/kubernetes/test/integration/apiserver/admissionwebhook: TestWebhookTimeoutWithoutWatchCache 
k8s.io/kubernetes/test/integration/auth: TestRBAC 
k8s.io/kubernetes/test/integration/client: TestSingleWatch 
k8s.io/kubernetes/test/integration/controlplane: TestObjectSizeResponses 
k8s.io/kubernetes/test/integration/deployment: TestDeploymentRollingUpdate 
```
Tests that failed in canary
```
k8s.io/kubernetes/test/integration/apiserver/admissionwebhook: TestWebhookTimeoutWithoutWatchCache 
k8s.io/kubernetes/test/integration/auth: TestServiceAccountTokenCreate 
k8s.io/kubernetes/test/integration/client: TestExecPluginRotationViaInformer 
k8s.io/kubernetes/test/integration/controlplane: TestKMSHealthz 
k8s.io/kubernetes/test/integration/deployment: TestDeploymentRollingUpdate 
k8s.io/kubernetes/test/integration/disruption: TestEmptySelector 
k8s.io/kubernetes/test/integration/disruption: TestEmptySelector/v1beta1_should_not_target_any_pods 
k8s.io/kubernetes/test/integration/disruption: TestEmptySelector/v1_should_target_all_pods 
```

The fact that `TestWebhookTimeoutWithoutWatchCache` shows up in both makes me think https://github.com/kubernetes/kubernetes/issues/103512 isn't resolved (on this branch?)

**Comment by spiffxp** [comment]
https://storage.googleapis.com/k8s-triage/index.html?ci=0&pr=1&job=pull-kubernetes-integration is failing to load in Chrome again, ref: https://github.com/kubernetes/test-infra/issues/22906

**Comment by ehashman** [comment]
> Another angle to take: why is the periodic / CI job passing where the presubmit is failing?
> 
>     * The CI job has a memory limit of 20Gi where the presubmit has a memory limit of 15Gi
> 
>     * The CI job runs with env `SHORT=--short=false`
> 
> 
> I would be more inclined to lend credence to these differences if I saw presubmit/periodic differences in other release branches

I diffed the yamls and I also noticed that the presubmit sets a context: https://github.com/kubernetes/test-infra/blob/76eeb12d9aa94f8507714ab2ce4d179b162dc5b9/config/jobs/kubernetes/sig-release/release-branch-jobs/1.22.yaml#L1780

The periodic doesn't and neither do previous release branches: https://github.com/kubernetes/test-infra/blob/76eeb12d9aa94f8507714ab2ce4d179b162dc5b9/config/jobs/kubernetes/sig-release/release-branch-jobs/1.21.yaml#L1508-L1517

Do we know why the 1.22 presubmits have this but no other jobs do?

**Comment by spiffxp** [comment]
I don't know off the top of my head but I suspect it would be highly unrelated, since it has to do with which github context to report to, ref: https://github.com/kubernetes/test-infra/blob/master/prow/config/jobs.go#L283

Probably something something config-forker / config-rotator serialize into/out-of prowjob structs, and somehow this got picked up as a now explicitly serialized thing along the way

**Comment by spiffxp** [comment]
At this point I'm having difficulty repro'ing locally, and am trying to narrow down the failure via https://github.com/kubernetes/kubernetes/pull/105488.  All kinds of force-pushes and stuff in there so it's noisy.  Summary thus far:
- tried reverting a few other PRs, still fails
- tried running just `make test-integration` in case `make test-cmd` was leaving behind detritus... fails faster
- currently: drop parallelism by setting `KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=1`

**Comment by spiffxp** [comment]
Perhaps obvious to those who know this code better, but I think the best signal for these failures so far is `client connection lost` showing up in the build log: https://storage.googleapis.com/k8s-triage/index.html?ci=0&pr=1&text=client%20connection%20lost&job=pull-kubernetes-integration

**Comment by aojea** [comment]
Disabling the podsecurity policy jobs the job goes through

https://github.com/kubernetes/kubernetes/pull/105506

but that gives more questions than answers though

@liggitt is it possible that the podsecurity tests logic to detect the version behaves differently depending on the job?

**Comment by liggitt** [comment]
> @liggitt is it possible that the podsecurity tests logic to detect the version behaves differently depending on the job?

no, that test just makes a lot of requests. It seems likely there's an issue between apiserver and etcd a high enough volume of requests exposes. Are the versions of etcd in master and 1.22 tests different?

**Comment by liggitt** [comment]
> * drop parallelism by setting `KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=1`, failed with a 357MB junit.xml

that's really surprising to me ... do you have a link to that run? the parallel runs/tests were what I saw exhausting ephemeral ports in the past

**Comment by spiffxp** [comment]
https://github.com/kubernetes/kubernetes/pull/105488/commits/5197769219f78c7fcd564fb834874cf32e1d4a73 - dropping concurrency down to 1 ended up with a 357MB (!!) junit xml file which spyglass is refusing to parse (as it should, that's way too large) (ref: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105488/pull-kubernetes-integration/1445512318795386880)

Grepping for ^FAIL in the log I pulled out:
```
test/integration/client TestExecPluginRotationViaInformer
test/integration/controlplane TestStatus
test/integration/pods TestPodEphemeralContainersDisabled
```

**Comment by liggitt** [comment]
looking at the stdout in the serial run, I see panic.go only in TestQuota:

```
grep panic.go serialrun.stdout 
{"Time":"2021-10-05T22:36:33.364456722Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:36:33.364376   64799 panic.go:965] \"HTTP\" verb=\"POST\" URI=\"/api/v1/namespaces/quotaed/pods\" latency=\"2m25.369478025s\" userAgent=\"quota.test/v0.0.0 (linux/amd64) kubernetes/$Format\" audit-ID=\"a6ad587a-1fe9-48f0-afba-248a92b86aae\" srcIP=\"127.0.0.1:41978\" resp=500 statusStack=\"\\ngoroutine 7281 [running]:\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc004351c80, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:274 +0xcf\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc004351c80, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:252 +0x35\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc004882200, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/a"}
{"Time":"2021-10-05T22:36:33.365107218Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:36:33.365019   64799 panic.go:965] \"HTTP\" verb=\"POST\" URI=\"/api/v1/namespaces/quotaed/pods\" latency=\"2m25.282338743s\" userAgent=\"quota.test/v0.0.0 (linux/amd64) kubernetes/$Format\" audit-ID=\"75de8fd4-2715-452c-9072-f6bf349befc7\" srcIP=\"127.0.0.1:41950\" resp=500 statusStack=\"\\ngoroutine 7320 [running]:\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc004e24100, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:274 +0xcf\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc004e24100, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:252 +0x35\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc004882300, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/a"}
{"Time":"2021-10-05T22:36:33.439372328Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:36:33.395482   64799 panic.go:965] \"HTTP\" verb=\"POST\" URI=\"/api/v1/namespaces/quotaed/pods\" latency=\"2m25.323783326s\" userAgent=\"quota.test/v0.0.0 (linux/amd64) kubernetes/$Format\" audit-ID=\"6ed4cc53-a55e-492c-8968-9ba4e0d628c5\" srcIP=\"127.0.0.1:41966\" resp=500 statusStack=\"\\ngoroutine 7317 [running]:\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc004e24000, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:274 +0xcf\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc004e24000, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:252 +0x35\\nk8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc004882280, 0x1f4)\\n\\t/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/a"}
```

and it looks like TestQuota output is 50% of the total output (~700mb):

```
grep TestQuota serialrun.stdout  | wc -c
 730919143
```

it looks like the resource quota controller goes nuts at about 22:33:39.331724:
```
{"Time":"2021-10-05T22:33:39.33173509Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331724   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
{"Time":"2021-10-05T22:33:39.331789854Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331759   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
{"Time":"2021-10-05T22:33:39.331807336Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331796   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
{"Time":"2021-10-05T22:33:39.331815556Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331807   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
{"Time":"2021-10-05T22:33:39.331824314Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331819   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
{"Time":"2021-10-05T22:33:39.331832705Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331828   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
{"Time":"2021-10-05T22:33:39.331918832Z","Action":"output","Package":"k8s.io/kubernetes/test/integration/quota","Test":"TestQuota","Output":"I1005 22:33:39.331850   64799 resource_quota_controller.go:194] Resource quota controller queued all resource quota for full calculation of usage\n"}
```

that's logged about 2.5 million times in about 3.5 minutes, apparently in a hotloop

**Comment by liggitt** [comment]
fwiw, from scraping stdout artifacts from last passing runs, tests with >10k output lines:

master:
```
15303 TestAdoption
16874 TestPreemptionRaces/ensures_that_other_pods_are_not_scheduled_while_preemptor_is_being_marked_as_nominated_(issue_#72124)
18485 TestConfigConsumerFight
18719 TestPersistentVolumeControllerStartup
26886 TestPersistentVolumeMultiPVsPVCs
29415 TestPersistentVolumeProvisionMultiPVCs
31166 TestVolumeBindingDynamicStressFast
41283 TestVolumeBindingDynamicStressSlow
41588 TestVolumeBindingStressWithSchedulerResync
43103 TestVolumeBindingStress
298114 TestQuotaLimitService
4181892 TestQuota
```

1.22:
```
13097 TestAdoption
15897 TestPersistentVolumeMultiPVsPVCs
16850 TestPreemptionRaces/ensures_that_other_pods_are_not_scheduled_while_preemptor_is_being_marked_as_nominated_(issue_#72124)
18218 TestPersistentVolumeControllerStartup
22433 TestPersistentVolumeProvisionMultiPVCs
24636 TestVolumeBindingDynamicStressFast
28755 TestVolumeBindingStress
33113 TestVolumeBindingDynamicStressSlow
38043 TestVolumeBindingStressWithSchedulerResync
144523 TestQuotaLimitService
1812887 TestQuota
```

**Comment by liggitt** [comment]
if test-integration is on the edge, I think it's more likely master is working because it dropped a lot of v1beta1 integration tests (https://github.com/kubernetes/kubernetes/pull/105506#issuecomment-936447807)

to unblock 1.22, I'd probably recommend switching the presubmit config to match the periodic config, which seems to be green (bump memory slightly, run all integration tests)

**Comment by spiffxp** [comment]
https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-integration-1-22-canary - OK I'm willing to port to the integration job with a comment linking to this issue to explain why 1.22 gets a different job config
